core/trino-main/src/main/java/io/trino/operator/PartitionedLookupSourceFactory.java

```diff
@@ -212,7 +212,10 @@ public void setPartitionSpilledLookupSourceHandle(int partitionIndex, SpilledLoo
 
         lock.writeLock().lock();
         try {
-            if (destroyed.isDone()) {
+            if (partitionedConsumption.isDone() || destroyed.isDone()) {
+                // Either:
+                // 1. join operators already finished consuming probe data and are unspilling
```

Comment:
perhaps the condition should be on `partitionsNoLongerNeeded.isDone()`
then you don't need `|| destroyed.isDone()` part anymore.

Reply:
> partitionsNoLongerNeeded is not setup in finishProbeOperator

it seems to be set here https://github.com/trinodb/trino/blob/03d8e5abc686c5c4a9f96ca14db07e2aed880174/core/trino-main/src/main/java/io/trino/operator/PartitionedLookupSourceFactory.java#L326

Reply:
> nope. They could have already processed all probe data because at that time HashBuilder was not spilled. It's not short-circuit.

if `setPartitionSpilledLookupSourceHandle` is being called, it seems to mean the `lookupSourceSupplier` was not set yet and than means Probe should be waiting here
on `lookupSourceProviderFuture.isDone()` being `false` in 
https://github.com/trinodb/trino/blob/4421ee68994e6c524882043b104ccb4d53e9312e/core/trino-main/src/main/java/io/trino/operator/LookupJoinOperator.java#L124

what am i missing?

Reply:
1. `lookupSourceProviderFuture` can unblock on non-spilled lookup source
2. This method is about `io.trino.operator.SpilledLookupSourceHandle`. I believe a new lookup source is created when spill happens.